src/Yesod/Auth/OAuth2.hs

```diff
@@ -51,3 +60,29 @@ authOAuth2Widget widget name oauth getCreds =
     AuthPlugin name (dispatchAuthRequest name oauth getCreds) login
   where
     login tm = [whamlet|<a href=@{tm $ oauth2Url name}>^{widget}|]
+
+-- | Read from the values set via @'setExtra'@
+--
+-- This is unsafe.
+--
+getAccessToken :: Creds m -> AccessToken
+getAccessToken = AccessToken
+    . fromJustNote "yesod-auth-oauth2 bug: credsExtra without accessToken"
```

Comment:
It's a bummer that we have to use `fromJust` here. Is this because `Yesod.Auth.Creds` doesn't have a type parameter, so we have to use stringly typed data for the "extra" bits? Would it be worth trying to change this in `Yesod.Auth`?

Reply:
Also, this is not necessarily a bug, right? If somebody authenticates with username/password and then calls `getAccessToken` with those creds, that will compile and crash?

Reply:
> Would it be worth trying to change this in Yesod.Auth?

I'm not really sure that would work. Even if it were `Creds e m`, for some extras type `e`, that `e` couldn't be plugin-specific because it must be the same across all plugins in use. Maybe I'm missing something clever though.

> Also, this is not necessarily a bug, right? If somebody authenticates with username/password and then...

You're absolutely right; great catch. I think I lean towards making this `Maybe` now. I thought there was strong guarantees a value would be present, but non-oauth2-plugins is way too likely.